### PR TITLE
Normalizes all of the "See also" blocks in comments.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description
This normalizes all of the "See also" blocks in comments so that they conform in style.

They all look like this now: 

```dart
/// See also:
///
///  * [MyFavoriteWidget], because it's cool.
class Foo {}
```

I removed some useless "See also" blocks, and added commentary to ones that were just "bare" references.

This is my penance for adding so many bullets that are "-" instead of "*". :-)

## Tests

- No tests, because these are only docs changes.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/diagnostics.dart [hunk 1]
```diff
@@ -512,7 +513,7 @@ final TextTreeConfiguration denseTextConfiguration = TextTreeConfiguration(
 ///    ╚═══════════
 /// ```
 ///
-/// /// See also:
+ /// See also:
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/diagnostics.dart]: nit: Looks like there may still be an extra space here at the beginning of the line?

File: packages/flutter/lib/src/foundation/key.dart [hunk 2]
```diff
@@ -17,7 +17,10 @@ import 'package:meta/meta.dart';
 ///
 /// Subclasses of [Key] should either subclass [LocalKey] or [GlobalKey].
 ///
-/// See also the discussion at [Widget.key].
+/// See also:
+///
+///  * The discussion at [Widget.key] for more information about how widgets use
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/key.dart]: This should probably be: "* [Widget.key], which discusses how widgets use keys." to be more in line with our style?

Comment (goderbauer) [reply packages/flutter/lib/src/foundation/key.dart]: Same below

File: packages/flutter/lib/src/rendering/box.dart [hunk 3]
```diff
@@ -1511,7 +1513,9 @@ abstract class RenderBox extends RenderObject {
   ///
   /// This function should never return a negative or infinite value.
   ///
-  /// See also examples in the definition of [computeMinIntrinsicWidth].
+  /// See also:
+  ///
+  ///  * Examples in the documentation for [computeMinIntrinsicWidth].
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/box.dart]: The thing to look at usually comes first: [foo], which is the bar.

**Review by goderbauer — APPROVED** [review]
RSLGTM

File: packages/flutter/lib/src/foundation/diagnostics.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/foundation/diagnostics.dart]: Oooh, good catch.  Fixed.

File: packages/flutter/lib/src/foundation/key.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/foundation/key.dart]: OK, fixed.

File: packages/flutter/lib/src/rendering/box.dart (hunk 3 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/rendering/box.dart]: Fixed, and below.

**Pull request merged**